src/hotspot/share/gc/z/zForwarding.hpp

```diff
@@ -140,10 +142,15 @@ class ZForwarding {
   void mark_done();
   bool is_done() const;
 
-  zaddress find(zaddress_unsafe addr);
-
   ZForwardingEntry find(uintptr_t from_index, ZForwardingCursor* cursor) const;
+  zaddress find(zoffset from_offset, ZForwardingCursor* cursor);
+  zaddress find(zaddress from_addr, ZForwardingCursor* cursor);
+  zaddress find(zaddress_unsafe from_addr, ZForwardingCursor* cursor);
+  zaddress find(zaddress_unsafe from_addr);
+
   zoffset insert(uintptr_t from_index, zoffset to_offset, ZForwardingCursor* cursor);
+  zaddress insert(zoffset from_offset, zaddress to_addr, ZForwardingCursor* cursor);
+  zaddress insert(zaddress from_addr, zaddress to_addr, ZForwardingCursor* cursor);
```

Comment:
Could some of these functions be moved to the `private:` section where you added the `index` function?

Reply:
I moved four methods.